rustls/src/conn.rs

```diff
@@ -189,6 +189,20 @@ mod connection {
                 (false, false) => Err(io::ErrorKind::WouldBlock.into()),
             }
         }
+
+        /// Obtain a chunk of plaintext data received from the peer over this TLS connection.
+        /// This does the same thing as [`Reader::fill_buf`], but unlike that method, this
+        /// method returns a slice whose lifetime is bounded by the [`ConnectionCommon`]
+        /// that created this `Reader`.
+        pub fn into_first_chunk(self) -> io::Result<&'a [u8]> {
```

Comment:
Why is this necessary/useful? It seems unrelated to the `ReadBuf` impl.

Reply:
It's necessary to be able to implement `tokio::io::AsyncBufRead` in `tokio_rustls`, since otherwise `rustls::Reader::fill_buf` returns a buffer with too short of a lifetime (the lifetime of the `Reader`, which is constructed temporarily in tokio_rustls - not of the `ConnectionCommon`). Neither method is strictly more useful than the other though.